Cron/KlSyncs.php

```diff
@@ -0,0 +1,187 @@
+     * @throws Exception
+     */
+    public function sync()
+    {
+        $syncCollection = $this->_syncCollectionFactory->create();
+        $groupedRows = $this->getGroupedRows($syncCollection->getRowsForSync('NEW')->getData());
+
+        $this->sendUpdatesToApp($groupedRows);
+
+        return;
+    }
+
+    /**
+     * Cleanup Cron job removing rows marked as SYNCED from kl_sync table
+     */
+    public function clean()
+    {
+        $syncCollection = $this->_syncCollectionFactory->create();
+        $idsToDelete = $syncCollection->getIdsToDelete('SYNCED');
+
+        $syncCollection->deleteRows($idsToDelete);
+
+        $klSyncTableSize = $syncCollection->count();
+        if ($klSyncTableSize > 10000 ){
+            $this->_klaviyoLogger->log("WARNING: kl_sync table size greater than 10000, currently sitting at $klSyncTableSize");
+        }
+
+        return;
+    }
+
+    /**
+     * Retry Cron job retries all rows marked for retry in kl_sync table
+     * @throws Exception
+     */
+    public function retry()
+    {
+        $syncCollection = $this->_syncCollectionFactory->create();
+        $groupedRows = $this->getGroupedRows($syncCollection->getRowsForSync('RETRY')->getData());
+
+        $this->sendUpdatesToApp($groupedRows, true);
+
+        return;
+    }
+
+    /**
+     * Sends Webhook or Track API requests based on the topic of each row
+     * and creates a response manifest to updates row status
+     * @param array $groupedRows
+     * @param bool $isRetry
+     * @throws Exception
+     */
+    private function sendUpdatesToApp(array $groupedRows, bool $isRetry = false)
+    {
+        $webhookTopics = ['product/save']; //List of topics that use webhooks
+        $trackApiTopics = ['Added To Cart']; //List of topics that use the Track API
+
+        $responseManifest = ['1' => [], '0' => []];
+
+        foreach($groupedRows as $topic => $rows){
+            if (in_array($topic, $webhookTopics) && !empty($rows)) {
+                foreach($rows as $row) {
+                    $response = $this->_webhookHelper->makeWebhookRequest(
+                        $row['topic'],
+                        $row['payload'],
+                        $row['klaviyo_id']
+                    );
+                    if (!$response) {$response = '0';}
+
+                    array_push( $responseManifest["$response"], $row['id']);
+                }
+            }
+
+            if (in_array($topic, $trackApiTopics) && !empty($rows)) {
+                foreach($rows as $row) {
+                    $response = $this->_dataHelper->klaviyoTrackEvent(
+                        $row['topic'],
+                        json_decode($row['user_properties'], true ),
+                        json_decode($row['payload'], true ),
+                        strtotime($row['created_at'])
+                    );
+                    if (!$response) {$response = '0';}
+
+                    array_push($responseManifest["$response"], $row['id']);
```

Comment:
The condition in which the klaviyoTrackEvent method will return that string is taken care of in the SalesQuoteSaveAfter where we check if the customer is identified, it is fairly unlikely for that string to be returned since a row in the DB will not exist without the customer properties